mlfinlab/backtest_statistics/backtests.py

```diff
@@ -0,0 +1,546 @@
+
+        # Arrays with adjusted Sharpe ratios and haircuts
+        sr_adj = np.zeros(4)
+        haircut = np.zeros(4)
+        # Adjusted Sharpe ratios and haircut percentages
+        sr_adj[0], haircut[0] = self._sharpe_ratio_haircut(p_val_adj[0], monthly_obs, sr_annual)
+        sr_adj[1], haircut[1] = self._sharpe_ratio_haircut(p_val_adj[1], monthly_obs, sr_annual)
+        sr_adj[2], haircut[2] = self._sharpe_ratio_haircut(p_val_adj[2], monthly_obs, sr_annual)
+        sr_adj[3], haircut[3] = self._sharpe_ratio_haircut(p_val_adj[3], monthly_obs, sr_annual)
+
+        results = np.array([p_val_adj,
+                            sr_adj,
+                            haircut])
+
+        return results
+
+    def profit_hurdle(self, num_mult_test, num_obs, alpha_sig, vol_anu, rho):
+        # pylint: disable=too-many-locals
+        """
+        Calculates the required mean monthly return for a strategy at a given level of significance.
+
+        This algorithm uses four adjustment methods - Bonferroni, Holm, BHY (Benjamini, Hochberg and Yekutieli)
+        and the Average of them. The result is the Minimum Average Monthly Return for the strategy to be significant
+        at a given significance level, taking into account multiple testing.
+
+        This function doesn't allow for any autocorrelation in the strategy returns.
+
+        :param num_mult_test: (int) Number of tests in multiple testing allowed (number of other strategies tested)
+        :param num_obs: (int) Number of monthly observations for a strategy
+        :param alpha_sig: (float) Significance level (e.g., 5%)
+        :param vol_anu: (float) Annual volatility of returns(e.g., 0.05 or 5%)
+        :param rho: (float) Average correlation among returns of strategies tested
+        :return: (np.ndarray) Minimum Average Monthly Returns for
+                              [Independent tests, Bonferroni, Holm, BHY and Average for Multiple tests]
+        """
+        # Independent test t-statistic
+        tstat_independent = ss.norm.ppf((1 - alpha_sig / 2), 0, 1)
+
+        # Bonferroni t-statistic
+        p_value_bonferroni = np.divide(alpha_sig, num_mult_test)
+        tstat_bonderroni = ss.norm.ppf((1 - p_value_bonferroni / 2), 0, 1)
+
+        # Estimating the parameters used for distributions based on HLZ model
+        # Result is [rho, n_simulations, prob_zero_mean, lambd]
+        parameters = self._parameter_calculation(rho)
+
+        # Needed number of trails inside a simulation with the check of (num_simulations >= num_mul_tests)
+        num_trails = int((np.floor(num_mult_test / parameters[1]) + 1) * np.floor(parameters[1] + 1))
+        # Generating a panel of t-ratios (of size self.simulations * num_simulations)
+        t_sample = self._sample_random_multest(parameters[0], num_trails, parameters[2], parameters[3],
+                                               self.simulations)
+
+        # Holm method
```

Comment:
I see that the code below for Holm and BHY method is almost the same except for the calls to self._holm_method_returns() and self._bhy_method_returns(). You can try to minimise the redundant code here by adding them in a single FOR loop.

Reply:
Did both methods in one FOR loop. Thank you.